Add name.variable and value.variable

pokey opened the pull request:
I think it's worth distinguishing between an assignment (`foo = bar`) and a variable declarateion (`const foo = bar`) to make it clear we want to support both. I think the name `name.variable` and `value.variable` are clear enough and consistent with `name.function`

## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [-] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [-] I have not broken the cheatsheet

AndreasArvidsson commented:
I don't see you're actually using both assignment and declaration here? You're renaming assignment to variable. Shouldn't we just keep assignment and add variable as an extra facet?
variable => `const foo = 2`
assignment => `foo = 2`

pokey commented:
For name and value I did keep assignment, and added declaration as additional. Maybe you missed it? For type I just renamed assignment because I don't think type makes sense for assignments

AndreasArvidsson commented:
Ah right.
